eng/versioning/pom_file_version_scanner.ps1

```diff
@@ -0,0 +1,370 @@
+                        Write-Output "$($retVal)"
+                    }
+                }
+            }
+            else 
+            {   
+                $script:FoundError = $true
+                # <!-- {x-version-update;<groupId>:<artifactId>;current} -->
+                # every project string needs to have an update tag and projects version tags are always 'current'
+                Write-Output "Error: Missing project/version update tag. The tag should be <!-- {x-version-update;$($groupId):$($artifactId);current} -->"
+            }
+            
+        }
+        else 
+        {
+            # output an error for missing version element
+            $script:FoundError = $true
+            Write-Output "Error: Could not find project/version node for $($pomFile)"
+        }
+    }
+
+    if ($xmlPomFile.parent) {
+        # Verify the parent's version
+        $versionNode = $xmlPomFile.SelectSingleNode("/ns:project/ns:parent/ns:version", $xmlNsManager)
+        if ($xmlPomFile.project.parent.version -and $versionNode)
+        {
+            $artifactId = $xmlPomFile.project.parent.artifactId
+            $groupId = $xmlPomFile.project.parent.groupId
+            # versionNode.NextSibling.Value should be the actual XML tag starting with {x-version-update
+            if ($versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment") 
+            {
+                # parent version
+                if ($versionNode.NextSibling.Value.Trim() -ne "{x-version-update;$($groupId):$($artifactId);current}")
+                {
+                    $script:FoundError = $true
+                    Write-Output "Error: project/parent/version update tag should be <!-- {x-version-update;$($groupId):$($artifactId);current} -->"
+                }
+                else
+                {
+                    # verify the version tag and version are correct
+                    $retVal = Test-Dependency-Tag-And-Version $libHash $extDepHash $versionNode.InnerText $versionNode.NextSibling.Value
+                    if ($retVal)
+                    {
+                        $script:FoundError = $true
+                        Write-Output "$($retVal)"
+                    }
+                }
+            }
+            else 
+            {   
+                $script:FoundError = $true
+                # every project string needs to have an update tag and projects version tags are always 'current'
+                Write-Output "Error: Missing project/parent/version update tag. The tag should be <!-- {x-version-update;$($groupId):$($artifactId);current} -->"
+            }
+        }
+        else 
+        {
+            # output an error for missing version element
+            $script:FoundError = $true
+            Write-Output "Error: Could not find project/parent/version node for $($pomFile)"
+        }    
+    }
+
+    # Verify every dependency as a group, artifact and version
```

Comment:
```suggestion
    # Verify every dependency has a group, artifact and version
```

Reply:
Fixed locally, will be updated with the next commit.